src/ess/spectroscopy/indirect/normalisation.py

```diff
@@ -105,23 +163,141 @@ def monitor_slowness(
+
+    # names = {slow: wavelength}
+    converted = monitor.transform_coords(incident_wavelength=converter)
+    return converted
+
+
+def monitor_sloth(
+    primary: PrimarySpectrometerObject, monitor: SlownessMonitor
+) -> SlothMonitor:
+    """Convert the independent 'slowness' coordinate of a histogram DataArray to the
+    equivalent sloth, which is -- equivalently -- normalised slowness, normalised
+    inverse velocity, or normalised (incident) wavelength
+
+    Parameters
+    ----------
+    primary:
+        The primary spectrometer object describing the choppers and guide setting(s)
+    monitor:
+        A histogram beam monitor which has been converted from recorded 'frame time' to
+        inverse neutron velocity
+
+    Returns
+    -------
+    :
+        The same intensities with independent axis converted to the sloth,
+        which is the normalised slowness, inverse velocity, and incident wavelength
+    """
+    from choppera.nexus import primary_slowness
+    from scipp import max, min
+
+    from ..utils import in_same_unit
+
+    slow = 'slowness'
+    if slow not in monitor.coords:
+        raise RuntimeError(f'A SlownessMonitor must have the coordinate "{slow}"')
+    sloth = 'sloth'
+    names = {slow: sloth}
+    min_max = in_same_unit(primary_slowness(primary), to=monitor.coords[slow])
+    normed = (
+        (monitor.coords[slow] - min(min_max)) / (max(min_max) - min(min_max))
+    ).rename(names)
+    return DataArray(monitor.data.rename(names), coords={sloth: normed})
+
+
+def normalise(
+    events: WavelengthEvents,
+    monitor: WavelengthMonitor,
+    edges: WavelengthBins,
+) -> NormWavelengthEvents:
+    """Ensure the WavelengthEvents are binned according to the WavelengthBins, then use
+    the WavelengthMonitor as a lookup table to record the per-bin normalization.
+
+    Parameters
+    ----------
+    events
+        Event data which includes a per-event sloth coordinate
+    monitor
+        (Probably) 1-D histogram data with a sloth coordinate
+    edges
+        1-D bin boundaries for the event data
+
+    Returns
+    -------
+    :
+        Event data binned by sloth, with a coordinate that is the per-bin normalization
+    """
+    from scipp import lookup
+
+    dim = 'incident_wavelength'
+    centres = (edges[:-1] + edges[1:]) / 2
+    variances = None
+    if monitor.variances is not None:
+        monitor = monitor.copy()
+        variances = monitor.copy()
+        variances.values = monitor.variances
+        monitor.variances = None
+        variances.variances = None
+    counts = lookup(monitor, dim)[centres]
+    if variances is not None:
+        # Bad form, maybe. But two events with the same sloth bin are normalized
+        # by the same monitor counts -- which has a known uncertainty -- and scipp
+        # refuses to allow lookup on data which has variances defined.
+        counts.variances = (lookup(variances, dim)[centres]).values
```

Comment:
This should probably use `UncertaintyBroadcastMode` from ESSreduce. But I don't know if the upper bound mode is applicable here.
Do you know how large the uncertainties of monitor compared to detector are? If they are small enough, we can drop them.

Reply:
Does the result from `broadcast_uncertainties` work with `scipp.loookup`? Because that was the issue I'm sidestepping here.
The monitor uncertainties are necessarily large at the minimal and maximal wavelengths transported by the guide, where the incident intensity goes to zero -- but the uncertainty relative to the detector uncertainty is probably always small due to the small cross section for inelastic scattering.

Reply:
> Does the result from broadcast_uncertainties work with scipp.loookup?

It should. We use it here: https://github.com/scipp/essdiffraction/blob/149203c29be6dd6e5dc49bf910015b308c0afa9e/src/ess/powder/correction.py#L56-L59

> The monitor uncertainties are necessarily large at the minimal and maximal wavelengths transported by the guide, where the incident intensity goes to zero -- but the uncertainty relative to the detector uncertainty is probably always small due to the small cross section for inelastic scattering.

I think this deserves some more investigation once we have the full workflow. We should ensure that we can ignore correlations around the edges of the data. Or avoid using those data.